kernel/aster-nix/src/vm/vmo/mod.rs

```diff
@@ -263,7 +264,17 @@ impl Vmo_ {
                     return Ok(committed_page.clone());
                 }
 
-                (clone_page(&committed_page)?, true)
+                if no_commit {
+                    (FrameAllocOptions::new(1).alloc_single()?, true)
+                } else {
+                    (clone_page(&committed_page)?, true)
+                }
+            } else if no_commit {
+                let page = FrameAllocOptions::new(1).alloc_single()?;
+                if let Some(pager) = &self.pager {
+                    pager.insert_page(cursor.index() as usize, page.clone())?;
+                }
```

Comment:
Wouldn't there be some robustness issues here? Here, the new page is directly written to the pager, but this VMO might still be a copy-on-write VMO, in which case it shouldn't affect the pager.

Reply:
You are right. And I have fixed this.

Reply:
Thank you for your correction. It is true that I overlooked the possible sharing situation earlier, and I have fixed this.